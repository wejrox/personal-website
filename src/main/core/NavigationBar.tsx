import * as React from "react";
import Row from "reactstrap/lib/Row";
import Col from "reactstrap/lib/Col";

class NavigationBar extends React.Component {

    public render() {
        return (
            <Row>
                <Col>
                    Navigation Bar
                </Col>
            </Row>
        );
    }
}

export default NavigationBar;