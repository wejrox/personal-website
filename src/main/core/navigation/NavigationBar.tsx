import * as React from "react";
import {Collapse, Container, Nav, Navbar, NavbarToggler, NavItem} from "reactstrap";
import {Link} from "react-router-dom";
import {isMobile} from "react-device-detect";
import {observer} from "mobx-react";
import {observable} from "mobx";
import {css} from "@emotion/core";

/**
 * Navigation bar fixed to the top of the page which remains visible as the user scrolls.
 */
@observer
class NavigationBar extends React.Component {
    
    // Whether or not the navigation bar is open.
    @observable public isOpen: boolean = false;
    navItemStyle = css`
        cursor: default !important;
        user-select: none !important;
        border-bottom: 2px solid rgba(97, 218, 251, 0) !important;
        transition: 500ms ease;
    
        &:hover {
            border-bottom: 2px solid rgba(97, 218, 251, 1) !important;
        }`;
    
    constructor(props: object) {
        super(props);
        
        this.toggle = this.toggle.bind(this);
    }
    
    public render() {
        return (
            <Navbar color="light" light sticky={"top"} expand={"md"}
                    className={"shadow-sm" + (isMobile ? "py-3" : "")}
                    css={css`padding: 0; width: 100vw;`}>
                <Container>
                    <Link to={"/"} className={"navbar-brand px-3"}>James McDowell</Link>
                    <NavbarToggler onClick={this.toggle} className={"mx-3"}/>
                    <Collapse isOpen={this.isOpen} navbar className={"justify-content-end"}>
                        <Nav navbar vertical={false}>
                            <NavItem onClick={this.toggle} css={this.navItemStyle}>
                                <Link className={"d-flex justify-content-center nav-link"} to={"/"}>Home</Link>
                            </NavItem>
                            <NavItem onClick={this.toggle} css={this.navItemStyle}>
                                <Link className={"d-flex justify-content-center nav-link"}
                                      to={"/projects/"}>Projects</Link>
                            </NavItem>
                            <NavItem onClick={this.toggle} css={this.navItemStyle}>
                                <Link className={"d-flex justify-content-center nav-link"}
                                      to={"/contact/"}>Contact</Link>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Container>
            </Navbar>
        );
    }
    
    /**
     * Toggles whether or not the navbar is open (mobile).
     */
    public toggle() {
        if (isMobile) {
            this.isOpen = !this.isOpen;
        }
    }
}

export default NavigationBar;
