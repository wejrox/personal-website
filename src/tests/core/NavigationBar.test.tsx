import * as Enzyme from "enzyme";
import * as React from "react";
import NavigationBar from "../../main/core/NavigationBar";

describe("NavigationBar Component", () => {
    const wrapper = Enzyme.shallow(<NavigationBar/>);

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});