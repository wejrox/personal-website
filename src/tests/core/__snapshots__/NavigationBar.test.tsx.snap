// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Component renders correctly 1`] = `ShallowWrapper {}`;
